.compile-contracts.txt

```diff
@@ -0,0 +1,6 @@
+This file is here as a marker for setup.py. This file is not being distributed
```

Comment:
Can we not just check if build/contracts.json exists?

Reply:
when working from a git checkout the contracts.json may exist, but you still want compilation to be done.

Reply:
the contract.json is not included in a git checkout

Reply:
you need to build it exectly when it is missing

Reply:
The current code in develop also does compile the contracts even if contracts.json is there. I think that's fine for a git checkout in case you're modifying the solidity files.

I don't need to build it in a sdist release.  If it's missing there for whatever reason, the install command should fail instead of trying to compile the contracts (with a possibly incompatible solidity compiler).

Reply:
I am not sure what you mean. If you install via git the file does not exist. Do you mean from a local copy? Then I would assume that it works as it works with the sdist and uses the already compiled file.

Reply:
If you're working with a locally checked out clone of the git repository and install that with `pip install .`, you end up with build/contracts.json.
If you then pull changes from the upstream git repository or change code yourself, and install with `pip install .`, you want the contracts to be compiled again - even though build/contracts.json already exists.

Reply:
> If you're working with a locally checked out clone of the git repository and install that with pip install ., you end up with build/contracts.json.
No you should not. It should not be created.

Reply:
They only get written into the venv. So I would assume if someone creates that file manually, he actually wants that file.

Reply:
ok. right for pip.

bdist_wheel will create it though. 

This is for people using the checked out source code, IMO they want recompilation to happen. you don't know how they will use this repo.

And the behaviour for the locally checked out source code has not been changed AFAICT. If there was already a bug in the develop branch, please open a bug report.

This PR is about building and uploading a sdist based release.